### PR TITLE
consider embedded files when calculating changed packages

### PR DESCRIPTION
**Pull request by bhcleek**

### CODE REVIEW

File: gta.go [hunk 1]
```diff
@@ -237,15 +237,31 @@ func (g *GTA) markedPackages() (map[string]map[string]bool, error) {
...
+	embeddedChanged := make(map[string]struct{})
 	onlyTestsAffected := make(map[string]struct{})
 	onlyTestPackagesChanged := make(map[string]struct{})
 	for abs, dir := range dirs {
 		// TODO(bc): handle changes to go.mod when vendoring is not being used.
 
-		// ignore deleted directories that did not contain files.
+		// Add packages that embed the files of dir.
+		for _, f := range dir.Files {
+			// An embedded file may:
```
Comment (jakelogemann) [inline gta.go:252]: Could you perhaps rephrase these list items? I think I understand the semantics, I just know I likely wont in a year or two. 😅

**Comment by bhcleek** [comment]
Tests are failing in CI, but not locally. I'm not sure why yet.

File: gta.go (hunk 1 above)
Comment (bhcleek) [reply gta.go:252]: What do you suggest?

**Comment by bhcleek** [comment]
@jakelogemann nothing has changed since your review other than inserting the word `in` on that list and adding the testdata that I'd failed to commit.

File: gta.go [hunk 2]
```diff
@@ -237,15 +237,31 @@ func (g *GTA) markedPackages() (map[string]map[string]bool, error) {
...
+			//   1. reside in a directory that contains go files and
+			//   2. reside in a directory that does not contain any go files
+			//   3. be embedded by multiple packages
+			// Therefore, do not try short-circuiting anything; just record that the
+			// embedding packages are changed.
```
Comment (jakelogemann) [inline gta.go:257]:
```suggestion
			// Note that an embedded file may reside in:
			//   1. a directory that contains other go files.
			//   2. a directory that does NOT contain any go files.
			//   3. multiple packages (simultaneously).
			//
			// Therefore, do not try short-circuiting anything; just record that the
			// embedding packages are changed.
```

File: gta.go (hunk 1 above)
Comment (jakelogemann) [reply gta.go:252]: Had [to suggest it separately](https://github.com/digitalocean/gta/pull/31/files#r949693088) ☹️

File: packager.go [hunk 3]
```diff
@@ -210,7 +228,7 @@ func resolveLocal(pkg *Package, dir string, modulesByDir map[string]string) {
 // module aware mode and flattened forward and reverse transitive dependency
 // graphs. When in GOPATH mode the map of directories to import paths will be
 // empty.
-func dependencyGraph(cfg *packages.Config, patterns []string) (moduleNamesByDir map[string]string, forward map[string]map[string]struct{}, reverse map[string]map[string]struct{}, err error) {
+func dependencyGraph(cfg *packages.Config, patterns []string) (moduleNamesByDir map[string]string, forward map[string]map[string]struct{}, reverse map[string]map[string]struct{}, packagesByEmbedFile map[string][]string, err error) {
```
Comment (nesv) [inline packager.go:231]: Maybe it's time for this function to return a struct. :joy:

**Review by nesv — APPROVED** [review]
LGTM

File: packager.go (hunk 3 above)
Comment (bhcleek) [reply packager.go:231]: Yeah, probably. I'll save that for a future change.

**Comment by bhcleek** [comment]
Rebased to get #33. No other changes were made since the last approval

**Review by nesv — APPROVED** [review]

**Pull request merged**